src/cpp/session/modules/SessionRCompletions.R

```diff
@@ -3323,6 +3323,7 @@ assign(x = ".rs.acCompletionTypes",
    .rs.makeCompletions(
       token = token,
       results = results.sorted,
-      packages = packages.sorted
+      packages = packages.sorted,
+      type = attr(results, "type")
```

Comment:
Note that completions are constructed from `results.sorted`, not `results` -- I think the `type` attribute needs to be extracted beforehand and sorted to ensure that the order matches the returned completion vector.